index.js

```diff
@@ -4,7 +4,22 @@ var chokidar = require("chokidar");
 
 var { Rcon } = require("rcon-client");
 
-var config = require("./config.json");
+var config;
+// Check if we have a local config file or a docker volume
+try {
+	if (fs.existsSync("./config/config.json")) {
+		config = require("./config/config.json");
```

Comment:
I wonder why you decided to make config a folder, isn't it fine to keep it in the root directory?

Reply:
Yes I can explain that:

Docker volumes are intended to bind to a folder (and not a single file). While it is possible to bind a single file this can be confusing. For example if you bind `/host/a.txt` to `/container/a.txt` and the file on the host does NOT exists then Docker wil l create a FOLDER called `a.txt`. Then it tries to start and you can get strange error messages. See for example the problems @ https://stackoverflow.com/questions/42248198/how-to-mount-a-single-file-in-a-volume

But technically it is possible it's just not best practice. I also thought about mounting the entire Factorio directory but that gives a bit more access then I'd like so I kept that to a single file.

Reply:
Hmm, alright.. My own Dockerfile for my bot just binds the entire bot's folder to the docker volume.. Any reasons to not use a `/src` folder and bind the files to it as well?

Reply:
Well, that works for development yes. But someone wanting to run the bot does not have (or needs) the source files. He just downloads/uses a pre-built docker image and only binds the config. (Or use env vars)

Reply:
True, I've never ran any docker images without using source files though. But having an entire `config` folder when the rest is in the root directory sounds weird to me. Guess it's time to move the files to `/src` and make a `/src/config` folder for it